### PR TITLE
Add an upgrade to the pilus to turn it into an injectisome

### LINKED ISSUE #4108: Add pilus upgrade to turn it into an injectisome (change damage type to toxic)
making a new damage type to take the variant into account is probably the cleanest way, that way this upgrade can also have a different sound for the attack

### PR DESCRIPTION
**Pull request by ender-titan1**
**Brief Description of What This PR Does**

This PR adds a new upgrade to the pilus to turn it into an injectisome, that deals toxin damage.
This replaces my previous PR and implements the feature in a much cleaner and better way, although I will note that this uses temporary graphics for the upgrade and a temporary sound for impaling.

**Related Issues**

Closes #4108 

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [X] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by buckly90** [comment]
Works perfectly well, nice job!

There are two issues I have noticed:

1. The MP cost is not being applied, and the upgrade is free.
2.You need to update the code comments to reflect that the pilus is dealing toxic and not physical damage. This is pretty minor though and honestly is just bugging me.

**Comment by ender-titan1** [comment]
Curious, in my testing, the upgrade cost is being applied and does remove your MP. However, I noticed that the MP used for upgrading an organelle will be returned to you even if not in the same editor session when reverting back to no upgrade, I'm pretty sure this is a bug. I also found that if you only add one pilus to your cell for some bizarre reason the microbe faces the opposite direction from my mouse.

**Comment by hhyyrylainen** [comment]
I guess the upgrade logic just compares the old price to the new price and refunds it. There should be an addition there to mark when an upgrade was created in the current editor session and can be refunded, and when it cannot be.

Can you share a screenshot of the one pilus cell? It might have something to do with https://github.com/Revolutionary-Games/Thrive/issues/3109 (and the more general suggested solution: https://github.com/Revolutionary-Games/Thrive/issues/3620)

**Comment by ender-titan1** [comment]
Here is the screenshot of the one pilus cell (I placed the pilus in the hex above the cytoplasm)
Screenshot

(For some reason it doesn't show my mouse, but it is there where the red dot is)

Should I add that extra check or should that be handled by a different PR?

**Comment by hhyyrylainen** [comment]
> Should I add that extra check or should that be handled by a different PR?

For the cost refund? It's probably better that it's a separate PR to keep things clean and get this merged faster. I'll at least open an issue about that problem tomorrow.

File: simulation_parameters/microbe_stage/organelles.json [hunk 1]
```diff
@@ -23,7 +23,20 @@
...
+      "none": {
+        "isDefault": true,
+        "name": "UPGRADE_NAME_NONE",
+        "description": "UPGRADE_DESCRIPTION_NONE"
+      },
+      "injectisome": {
+        "name": "UPGRADE_PILUS_INJECTISOME",
+        "description": "UPGRADE_PILUS_INJECTISOME_DESCRIPTION",
+        "mpCost": 25,
+        "iconPath": "res://assets/textures/gui/bevel/parts/ToxinVacuoleIcon.png"
```
Comment (hhyyrylainen) [inline simulation_parameters/microbe_stage/organelles.json:37]: We'll probably want a custom icon for this upgrade, but that can be done later.

File: src/microbe_stage/organelle_components/PilusComponent.cs [hunk 2]
```diff
@@ -45,7 +47,8 @@ public override void OnShapeParentChanged(Microbe newShapeParent, Vector3 offset
 
             // New ownerId
             var ownerId = currentShapesParent.CreateNewOwnerId(newShapeParent, transform, addedChildShapes[0]);
-            newShapeParent.AddPilus(ownerId);
+            var isInjectisome = organelle.Upgrades?.UnlockedFeatures.Contains(PILUS_INJECTISOME_UPGRADE_NAME);
+            newShapeParent.AddPilus(ownerId, isInjectisome ?? false);
```
Comment (hhyyrylainen) [inline src/microbe_stage/organelle_components/PilusComponent.cs:51]:
```suggestion
            newShapeParent.AddPilus(ownerId, isInjectisome == true);
```

This is my preferred way for checking if a nullable bool is set.

File: src/microbe_stage/organelle_components/PilusComponent.cs [hunk 3]
```diff
@@ -124,9 +127,11 @@ private void CreateShape(Microbe parent)
         shape.Radius = pilusSize / 10.0f;
         shape.Height = pilusSize;
 
+        var isInjectisome = organelle.Upgrades?.UnlockedFeatures.Contains(PILUS_INJECTISOME_UPGRADE_NAME);
+
         var ownerId = parent.CreateShapeOwner(shape);
         parent.ShapeOwnerAddShape(ownerId, shape);
-        parent.AddPilus(ownerId);
+        parent.AddPilus(ownerId, isInjectisome ?? false);
```
Comment (hhyyrylainen) [inline src/microbe_stage/organelle_components/PilusComponent.cs:134]: Same here.

**Review by hhyyrylainen — APPROVED** [review]
The code changes look good now. Just need to fix the translation merge conflicts before this can be merged.

**Pull request merged**